samcli/commands/_utils/template.py

```diff
@@ -261,3 +263,25 @@ def get_template_artifacts_format(template_file):
             for resource_id, resource in template_dict.get("Resources", {}).items()
         }.values()
     )
+
+
+def get_template_function_resource_ids(template_file, artifact):
+    """
+    Get a list of function logical ids from template file.
+    Function resource types include
+        AWS::Lambda::Function
+        AWS::Serverless::Function
+    :param template_file: template file location.
+    :param artifact: artifact of type IMAGE or ZIP
+    :return: list of artifact formats
+    """
+
+    template_dict = get_template_data(template_file=template_file)
+    _function_resource_ids = []
+    for resource_id, resource in template_dict.get("Resources", {}).items():
+        if resource.get("Properties", {}).get("PackageType", ZIP) == artifact and resource.get("Type") in [
```

Comment:
There was an existing function collector, which is been used in build command, can we use that one instead?

Reply:
does that process the template? I will take a look at it.

Reply:
its very specific to build, we would need some refactoring to make it more generic so that we can share that.

Reply:
I would still use `SamFunctionProvider` https://github.com/aws/aws-sam-cli/blob/develop/samcli/lib/providers/sam_function_provider.py#L16 which will handle parameter overrides etc.

Reply:
There are some circular dependencies that we need to take care of here, I vote we do this in a separate PR.